testing/test_skipping.py

```diff
@@ -148,16 +150,16 @@ def test_func(self):
                     pass
         """
         )
-        item.config._hackxyz = 3
+        item.config._hackxyz = 3  # type: ignore[attr-defined]
         skipped = evaluate_skip_marks(item)
         assert skipped
         assert skipped.reason == "condition: config._hackxyz"
 
 
 class TestXFail:
     @pytest.mark.parametrize("strict", [True, False])
-    def test_xfail_simple(self, testdir, strict):
-        item = testdir.getitem(
+    def test_xfail_simple(self, pytester, strict) -> None:
```

Comment:
Would be great to catch the `pytester`s that appear in the middle of the parameter list, like here. A search/replace of ` pytest, ` with ` pytester: Pytester, ` should do the trick (also for the rest of the file). Also in this case case can annotate `strict`, like so:

```suggestion
    def test_xfail_simple(self, pytester: Pytester, strict: bool) -> None:
```

Reply:
Argh, sorry for missing those! I added typehinting for the remaining `pytester` args, as well as for the `strict` args throughout.